api/envoy/extensions/filters/http/credential_injector/v3/credential_injector.proto

```diff
@@ -0,0 +1,106 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.credential_injector.v3;
+
+import "envoy/config/core/v3/http_uri.proto";
+import "envoy/extensions/transport_sockets/tls/v3/secret.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.credential_injector.v3";
+option java_outer_classname = "CredentialInjectorProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/credential_injector/v3;credential_injectorv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Credential Injector]
+// [#not-implemented-hide:]
+// Credential Injector :ref:`configuration overview <config_http_filters_credential_injector>`.
+// [#extension: envoy.filters.http.credential_injector]
+
+// Filter config.
+message CredentialInjector {
+  // Credential to inject into proxied requests.
+  Credential credential = 1 [(validate.rules).message = {required: true}];
+}
+
+// Credential to inject into proxied requests.
+//
+message Credential {
+  oneof credential_type {
+    option (validate.required) = true;
+
+    // Basic Auth Credential to inject.
+    BasicAuthCredential basic = 1;
+
+    // Bearer Token Credential to inject.
+    transport_sockets.tls.v3.SdsSecretConfig bearer = 2;
```

Comment:
Might be worth creating a message wrapping SdsSecretConfig for consistency and because we might need more fields later.